src/gov/nist/javax/sdp/parser/OriginFieldParser.java

```diff
@@ -59,11 +59,11 @@ public OriginField originField() throws ParseException {
             Token sessionId = lexer.getNextToken();
             // guard against very long session IDs
             String sessId = sessionId.getTokenValue();
-            if (sessId.length() > 18)
-                sessId = sessId.substring(sessId.length() - 18);
             try {
                 originField.setSessId(Long.parseLong(sessId));
             } catch (NumberFormatException ex) {
+                if (sessId.length() > 18)
```

Comment:
why dont we simply leave this validation where it was, and change literal to "19"?
If the provided id is not numeric, we will have a limitation of 18, but if its numeric the limitation will be imposed by Long==19, this is a bit inconsistent. wdyt?